sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/GpuTypeShims.scala

```diff
@@ -0,0 +1,270 @@
+      case _: YearMonthIntervalType => true
+      case _: DayTimeIntervalType => true
+      case _ => false
+    }
+  }
+
+  /**
+   * Convert the given value to Scalar
+   */
+  def toScalarForType(t: DataType, v: Any): Scalar = {
+    t match {
+      case _: YearMonthIntervalType => v match {
+        case i: Int => Scalar.fromInt(i)
+        case _ => throw new IllegalArgumentException(s"'$v: ${v.getClass}' is not supported" +
+            s" for IntType, expecting int")
+      }
+      case _: DayTimeIntervalType => v match {
+        case l: Long => Scalar.fromLong(l)
+        case _ => throw new IllegalArgumentException(s"'$v: ${v.getClass}' is not supported" +
+            s" for LongType, expecting Long")
+      }
+      case _ =>
+        throw new RuntimeException(s"Can not convert $v to scalar for type $t.")
+    }
+  }
+
+  def supportCsvRead(dt: DataType) : Boolean = {
+    dt match {
+      case DayTimeIntervalType(_, _) => true
+      case _ => false
+    }
+  }
+
+  def csvRead(cv: cudf.ColumnVector, dt: DataType): cudf.ColumnVector  = {
+    dt match {
+      case d: DayTimeIntervalType => GpuIntervalUtils.castStringToDTInterval(cv, d)
+      case _ => throw new RuntimeException(s"Not support type $dt.")
+    }
+  }
+
+  /**
+   * Whether the Shim supports day-time interval type for specific operator
+   * Alias, Add, Subtract, Positive... operators support day-time interval type
+   */
+  def isSupportedDayTimeType(dt: DataType): Boolean = dt.isInstanceOf[DayTimeIntervalType]
+
+  /**
+   * Whether the Shim supports year-month interval type
+   * Alias, Add, Subtract, Positive... operators support year-month interval type
+   */
+  def isSupportedYearMonthType(dt: DataType): Boolean = dt.isInstanceOf[YearMonthIntervalType]
+
+  /**
+   * Get additional arithmetic supported types for this Shim
+   */
+  def additionalArithmeticSupportedTypes: TypeSig = TypeSig.ansiIntervals
+
+  /**
+   * Get additional predicate supported types for this Shim
+   */
+  def additionalPredicateSupportedTypes: TypeSig = TypeSig.DAYTIME
+
+  /**
+   * Get additional Csv supported types for this Shim
+   */
+  def additionalCsvSupportedTypes: TypeSig = TypeSig.DAYTIME
+
+  def typesDayTimeCanCastTo: TypeSig = TypeSig.DAYTIME + TypeSig.STRING + TypeSig.integral
+
+  def typesYearMonthCanCastTo: TypeSig = TypeSig.integral
+
+  def typesDayTimeCanCastToOnSpark: TypeSig = TypeSig.DAYTIME + TypeSig.STRING + TypeSig.integral
+
+  def typesYearMonthCanCastToOnSpark: TypeSig = TypeSig.YEARMONTH + TypeSig.STRING +
+      TypeSig.integral
+
+  def additionalTypesBooleanCanCastTo: TypeSig = if (SQLConf.get.ansiEnabled) {
+    TypeSig.none
+  } else {
+    TypeSig.TIMESTAMP
+  }
+
+  def additionalTypesDateCanCastTo:TypeSig = if (SQLConf.get.ansiEnabled) {
+    TypeSig.none
+  } else {
+    TypeSig.BOOLEAN
+  }
+
+  def additionalTypesTimestampCanCastTo: TypeSig = if (SQLConf.get.ansiEnabled) {
+    TypeSig.none
+  } else {
+    TypeSig.BOOLEAN
+  }
```

Comment:
This code is new in this PR. The rest of this file is copied from the 330 shim version.

Reply:
Thanks. That makes sense.

Reply:
I have pushed new commits to resolve this.